lib/videoroom/meeting.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Videoroom.Meeting do
+  @moduledoc """
+  A Module responsible for handling a room.
+  """
+  use GenServer
+
+  alias Jellyfish.ServerMessage.PeerCrashed
+  alias Jellyfish.ServerMessage.PeerDisconnected
+  alias Jellyfish.ServerMessage.RoomCrashed
+
+  alias Jellyfish.Room
+
+  @room_table :room_table
+
+  # Api
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args,
+      name: {:via, Registry, {Videoroom.Registry, args[:name]}}
+    )
+  end
+
+  @spec add_peer(pid()) :: {:ok, binary()} | {:error, binary()}
+  def add_peer(meeting) do
+    GenServer.call(meeting, {:add_peer})
+  end
+
+  # Callbacks
+
+  @impl true
+  def init(args) do
```

Comment:
I'd unwrap the args as soon as possible and didn't pass the args around. You should check if all args are present (now the meeting will be using `nil` as a name)